K8SPXC-283 mention storage size in backup docs (#414)
percona/percona-xtradb-cluster-operator: source/backups.rst

Percona XtraDB Cluster Operator allows doing cluster backup in two ways.
*Scheduled backups* are configured in the
`deploy/cr.yaml <https://github.com/percona/percona-xtradb-cluster-operator/blob/master/deploy/cr.yaml>`_
file to be executed automatically in proper time. *On-demand backups*
can be done manually at any moment.

> Backup images are usually stored on `Amazon S3 or S3-compatible

Reviewer: ```suggestion
Backup files are usually stored on `Amazon S3 or S3-compatible
```

storage <https://en.wikipedia.org/wiki/Amazon_S3#S3_API_and_competing_services>`_,

> but storing backups on a private volume is also possible.

Reviewer: ```suggestion
but storing backups on Persistent Volumes is also possible.
```

please, s/private volume/Persistent Volumes/ everywhere.
we use one volume per one backup so the plural form is better - Volumes

> Here is an example which uses Amazon S3 storage for backups:

Reviewer: ```suggestion
Here is an example of `deploy/cr.yaml <https://github.com/percona/percona-xtradb-cluster-operator/blob/master/deploy/cr.yaml>` which uses Amazon S3 storage for backups:
```

Here is an example which uses Amazon S3 storage for backups:

.. code:: yaml

   ...

>    backup:

Reviewer: line 63 - `version: 0.3.0` should be removed

`Operator
Options <https://percona.github.io/percona-xtradb-cluster-operator/configure/operator>`__.

.. _backups-manual:

Making on-demand backup
-----------------------

To make an on-demand backup, the user should first configure the backup storage
in the ``backup.storages`` subsection of the ``deploy/cr.yaml`` configuration
file in a same way it was done for scheduled backups. When the
``deploy/cr.yaml`` file contains correctly configured storage and is applied
with ``kubectl`` command, use *a special backup configuration YAML file* with
the following contents:

> * **backup name** in the ``name`` key,
> * **PXC Cluster name** in the ``pxcCluster`` key,
> * **storage name** in the ``storageName`` key.

Reviewer: ```suggestion
* **backup name** in the ``metadata.name`` key,
* **PXC Cluster name** in the ``spec.pxcCluster`` key,
* **storage name** from ``deploy/cr.yaml`` in the ``spec.storageName`` key.
```

`Operator
Options <https://percona.github.io/percona-xtradb-cluster-operator/configure/operator>`__.

.. _backups-manual:

Making on-demand backup
-----------------------

To make an on-demand backup, the user should use a special backup configuration
YAML file with the following contents:

* **backup name** in the ``name`` key,
* **PXC Cluster name** in the ``pxcCluster``key,
* **storage name** in the ``storageName`` key.

> .. note:: The atual backup storage configuration is done in the ``backup``

Reviewer: ```suggestion
.. note:: The actual backup storage configuration is done in the ``backup``
```

   section of the ``deploy/cr.yaml`` configuration file.

The example of the backup configuration file is `deploy/backup/backup.yaml <https://github.com/percona/percona-xtradb-cluster-operator/blob/master/deploy/backup/backup.yaml>`_.

> When the backup config file is ready, actual backup command is executed:

Reviewer: ```suggestion
When the backup destination is configured and applied with `kubectl apply -f deploy/cr.yaml` command, the actual backup command is executed:
```

.. code:: bash

   kubectl apply -f deploy/backup/backup.yaml

.. note:: Storing backup settings in a separate file can be replaced by
   passing its content to the ``kubectl apply`` command as follows:

   .. code:: bash

      cat <<EOF | kubectl apply -f-
      apiVersion: pxc.percona.com/v1
      kind: PerconaXtraDBClusterBackup
      metadata:
        name: backup1
      spec:
        pxcCluster: cluster1
        storageName: fs-pvc
      EOF

.. _backups-private-volume:

> Storing backup on a private volume

Reviewer: should we give an example for S3 config?
should we explain `deploy/cr.yaml` changes before `deploy/backup/backup.yaml` changes?

`Operator
Options <https://percona.github.io/percona-xtradb-cluster-operator/configure/operator>`__.

.. _backups-manual:

Making on-demand backup
-----------------------

To make an on-demand backup, the user should first configure the backup storage
in the ``backup.storages`` subsection of the ``deploy/cr.yaml`` configuration

> file in a same way it was done for scheduled backups. Wneh the

Reviewer: ```suggestion
file in the same way it was done for scheduled backups. When the
```

``deploy/cr.yaml`` file contains correctly configured storage and is applied

> with ``kubectl`` commanad, use *a special backup configuration YAML file* with

Reviewer: ```suggestion
with ``kubectl`` command, use *a special backup configuration YAML file* with
```

     storageName: fs-pvc
   EOF

.. _backups-private-volume:

Storing backup on a private volume
-----------------------------------

Here is an example of the backup section fragment, which configures a private

> volume for filesystem-type storage:

Reviewer: I think `Making on-demand backup` section should be enhanced

2. Now find out correct names for the backup and the cluster. Available
   backups can be listed with the following command:

   .. code:: bash

      kubectl get pxc-backup

   And the following command will list available clusters:

   .. code:: bash

      kubectl get pxc

> 3. When both correct names are known, the actual restoration process can
>    be started as follows:

Reviewer: ```suggestion
3. When both correct names are known, it is needed to fix ``spec.pxcCluster`` and ``spec. backupName`` fields in ``deploy/backup/restore.yaml`` file and the actual restoration process can
   be started as follows:
```

by the following command:

.. code:: bash

   kubectl get pxc-backup

When the name is known, backup can be deleted as follows:

.. code:: bash

   kubectl delete pxc-backup/<backup-name>

.. _backups-copy:

Copy backup to a local machine
------------------------------

Make a local copy of a previously saved backup requires not more than
the backup name. This name can be taken from the list of available
backups returned by the following command:

.. code:: bash

   kubectl get pxc-backup

When the name is known, backup can be downloaded to the local machine as
follows:

.. code:: bash

   ./deploy/backup/copy-backup.sh <backup-name> path/to/dir

For example, this downloaded backup can be restored to the local
installation of Percona Server:

.. code:: bash

   service mysqld stop
   rm -rf /var/lib/mysql/*
   cat xtrabackup.stream | xbstream -x -C /var/lib/mysql
   xtrabackup --prepare --target-dir=/var/lib/mysql
   chown -R mysql:mysql /var/lib/mysql
   service mysqld start

.. note:: There is also a possibility to store backups on your local filesystem
   instead of remote storage. See :ref:`local.storage.example.backups` for

>    details on configuring this option.

Reviewer: please remove
